source/adapters/level_zero/event.cpp

```diff
@@ -168,7 +168,8 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueEventsWaitWithBarrier(
         // TODO: this and other special handling of in-order queues to be
         // updated when/if Level Zero adds native support for in-order queues.
         //
-        if (Queue->isInOrderQueue() && InOrderBarrierBySignal) {
+        if (Queue->isInOrderQueue() && InOrderBarrierBySignal &&
+            !Queue->isProfilingEnabled()) {
```

Comment:
It's not clear to me how profiling affects synchronization.

Reply:
@smaslov-intel : please see here 

https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=appendsignalevent#_CPPv430zeCommandListAppendSignalEvent24ze_command_list_handle_t17ze_event_handle_t

> zeCommandListAppendSignalEvent
>
> The duration of an event created from an event pool that was created using [ZE_EVENT_POOL_FLAG_KERNEL_TIMESTAMP](https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=appendsignalevent#ze__api_8h_1a148be5a8b03e9d401a415d2227cf9f11af1bab0559057f7ec51e2c77a1746894f) or [ZE_EVENT_POOL_FLAG_KERNEL_MAPPED_TIMESTAMP](https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=appendsignalevent#ze__api_8h_1a148be5a8b03e9d401a415d2227cf9f11a853fca9e461e0b8ae1ad7947476c9795) flags is undefined

Therefore,, when profiling is used, we cannot just used signalEvent if EventWaitList.Lenght == 0. We need to fallback directly to barrier to have correct timestamps.

Reply:
OK, but please add source code comment explaining this.

Reply:
thanks @smaslov-intel. I have added comment.